src/app/api/services/edit-page/edit-page.service.ts

```diff
@@ -3,22 +3,151 @@ import { Injectable } from '@angular/core';
 
 /**
  * Provide util methods to get a edit page html
  * @export
  * @class EditPageService
  */
+
 @Injectable()
 export class EditPageService {
     constructor(private coreWebService: CoreWebService) {}
 
-    get(url: string): Observable<DotRenderedPage> {
+    /**
+     * Get the page HTML in edit mode
+     *
+     * @param {string} url
+     * @returns {Observable<DotRenderedPage>}
+     * @memberof EditPageService
+     */
+    getEdit(url: string): Observable<DotRenderedPage> {
+        return this.get(url, 'EDIT_MODE');
+    }
+
+    /**
+     * Get the page HTML in preview mode
+     *
+     * @param {string} url
+     * @returns {Observable<DotRenderedPage>}
+     * @memberof EditPageService
+     */
+    getPreview(url: string): Observable<DotRenderedPage> {
+        return this.get(url, 'PREVIEW_MODE');
+    }
+
+    /**
+     * Get the page HTML in live mode
+     *
+     * @param {string} url
+     * @returns {Observable<DotRenderedPage>}
+     * @memberof EditPageService
+     */
+    getLive(url: string): Observable<DotRenderedPage> {
+        return this.get(url, 'LIVE_MODE');
+    }
+
+    /**
+     * Lock a content asset
+     *
+     * @param {string} inode
+     * @returns {Observable<any>}
+     * @memberof PageViewService
+     */
+    lock(inode: string): Observable<any> {
+        return this.coreWebService.requestView({
+            method: RequestMethod.Put,
+            url: `content/lock/inode/${inode}`
+        });
+    }
+
+    /**
+     * Unlock a content asset
+     *
+     * @param {string} inode
+     * @returns {Observable<any>}
+     * @memberof PageViewService
+     */
+    unlock(inode: string): Observable<any> {
+        return this.coreWebService.requestView({
+            method: RequestMethod.Put,
+            url: `content/unlock/inode/${inode}`
+        });
+    }
+
+    /**
+     * Set the page state
+     *
+     * @param {DotRenderedPage} page
+     * @param {DotEditPageState} state
+     * @returns {Observable<any>}
+     * @memberof EditPageService
+     */
+    setPageState(page: DotRenderedPage, state: DotEditPageState): Observable<DotRenderedPageState> {
+
+        const methods = [];
+        let lockUnlock;
+        let pageMode;
+
+        // When lock prop is not set is null
+        if (state.lock === true) {
+            lockUnlock = this.lock(page.liveInode)
+                .pluck('bodyJsonObject')
+                .pluck('message');
+        } else if (state.lock === false) {
+            lockUnlock = this.unlock(page.liveInode)
+                .pluck('bodyJsonObject')
+                .pluck('message');
+
+        }
+
+        if (state.mode === PageMode.PREVIEW) {
+            pageMode = this.getPreview(page.pageUri);
+        } else if (state.mode === PageMode.EDIT) {
+            pageMode = this.getEdit(page.pageUri);
+        } else if (state.mode === PageMode.LIVE) {
+            pageMode = this.getPreview(page.pageUri);
+        }
```

Comment:
we should use this method: 'private get(url: string, mode: string)' here

Reply:
This looks like state machine and
this could be handle by Map< PageMode, Method> map

map [PageMode.LIVE] = function (pageUri) {
      this.getEdit(pageUri)
}
....

then

pageMode = map[state.mode].(page.pageUri);

Reply:
I like that @jdotcms forgot about that pattern, done.